modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +71,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, error) {
+	if !ut.IsValid() {
+		return true, errorsmod.Wrap(ErrInvalidUpgradeTimeout, "upgrade timeout cannot be empty")
+	}
+
+	selfHeight, timeoutHeight := clienttypes.GetSelfHeight(ctx), ut.Height
+	if selfHeight.GTE(timeoutHeight) && timeoutHeight.GT(clienttypes.ZeroHeight()) {
```

Comment:
I would switch this ordering:
```suggestion
	if !timeoutHeight.IsZero() && selfHeight.GTE(timeoutHeight)  {
```